api_app/script_analyzers/observable_analyzers/stratosphere.py

```diff
@@ -0,0 +1,140 @@
+
+        with open(db_loc1, "r") as f:
+            db = f.read()
+        db_list = db.split("\n")
+        count = 0
+        for ip_tuple in db_list:
+            if count < 2:
+                count += 1
+                continue
+            else:
+                if ip in ip_tuple:
+                    ip_rating = ((ip_tuple.split(","))[2]).strip()
+                    result["new_attacker"] = True
+                    result["new_attacker_rating"] = ip_rating
+
+        with open(db_loc2, "r") as f:
+            db = f.read()
+        db_list = db.split("\n")
+        count = 0
+        for ip_tuple in db_list:
+            if count < 2:
+                count += 1
+                continue
+            else:
+                if ip in ip_tuple:
+                    ip_rating = ((ip_tuple.split(","))[2]).strip()
+                    result["repeated_attacker"] = True
+                    result["repeated_attacker_rating"] = ip_rating
+
+        return result
+
+    def updater(self):
+        try:
+            logger.info("starting download of dataset from stratosphere")
+
+            base_url = "https://mcfp.felk.cvut.cz"
+            mid_url = "/publicDatasets/CTU-AIPP-BlackList/Todays-Blacklists/"
+            url0 = base_url + mid_url + "AIP_blacklist_for_IPs_seen_last_24_hours.csv"
+            priority_url = "AIP_historical_blacklist_prioritized_by_"
+            url1 = base_url + mid_url + priority_url + "newest_attackers.csv"
+            url2 = base_url + mid_url + priority_url + "repeated_attackers.csv"
+
+            p = requests.get(url0, verify=False)
+            p.raise_for_status()
+
+            q = requests.get(url1, verify=False)
+            q.raise_for_status()
+
+            r = requests.get(url2, verify=False)
+            r.raise_for_status()
+
+            with open(db_loc0, "w") as f:
+                f.write(p.content.decode())
+
+            with open(db_loc1, "w") as f:
+                f.write(q.content.decode())
+
+            with open(db_loc2, "w") as f:
+                f.write(r.content.decode())
+
+            if not os.path.exists(db_loc0 or db_loc1 or db_loc2):
+                raise AnalyzerRunException("failed extraction of stratosphere dataset")
+
+            logger.info("ended download of dataset from stratosphere")
+
+        except Exception as e:
+            traceback.print_exc()
+            logger.exception(e)
+
+        db_location = [db_loc0, db_loc1, db_loc2]
+
+        return db_location
+
+    def check_dataset_status(self):
+        if not os.path.isfile(db_loc0 and db_loc1 and db_loc2):
+            self.updater()
+        today = date.today()
+
+        timestamp = os.path.getctime(db_loc0)
+        dt_object = datetime.fromtimestamp(timestamp)
+
+        if (
+            today.day == dt_object.day
+            and today.month == dt_object.month
+            and today.year == dt_object.year
+        ):
+            logger.info("Dataset is up to date")
+        else:
```

Comment:
This is ok but the problem is that these blacklists are updated usually some hours after the start of the day (UTC). This would mean that at the start of the new day, this would download the blacklist of the day before because it was not already updated.
If possible, I would add a reasonable hour when we expect those blacklists are already been updated and force the updater there. I think that 4 am UTC could be a reasonable date. (I saw that they seem to update those lists between 1 am and 2 am)

Reply:
Changed it to verify whether the file is downloaded before or after 4 am. If downloaded before 4 am it would force the updater to download again.